utils/common/src/main/java/brooklyn/util/collections/CollectionFunctionals.java

```diff
@@ -80,6 +80,21 @@ public Integer get() {
         @Override public String toString() { return "sizeSupplier("+collection+")"; }
     }
 
+    private static final class SizeSupplierFromSupplier implements Supplier<Integer> {
```

Comment:
Too many `Supplier`s make my head hurt! Minor naming nitpick to clarify: `SizeSupplierFromIterableSupplier` (adds `Iterable`).